### DISCUSSION TITLE
Use d3d11va2 instead of d3d11va?

### DISCUSSION THREAD

**Discussion by lukasf** [ffmpeginteropx/FFmpegInteropX#394] [General]
Just by coincidence, I stubled about the fact that there is not only d3d11va (which we use), but also d3d11va2. It is nowhere documented what the differences are. Actually, I found almost nothing on the web about this, except for a post from 2017, where someone says "d3d11va is old API - d3d11va2 is badly named new API, which you should use".

Just posting this as a reminder here. If we happen to have some time, we might look into this.

**Comment by brabebhin** [comment]
Does this mean we need to turn on a different flag in ffmpeg build? 

Usually in directx the later numbered versions add new features without breaking backwards compatibility. This is done to support new hardware features. We might already be using it, we have hevc and av1 support so i think we correctly use all hardware features. 

I think for example we actually use Id3d11device4 implementation for our Id3d11device pointers. You can assign device4 instead of device at any point and not worry about breaking the code.
